samza-sql/src/main/java/org/apache/samza/sql/translator/ScanTranslator.java

```diff
@@ -80,33 +86,69 @@ public boolean apply(KV<Object, Object> message) {
     this.queryId = queryId;
   }
 
+  /**
+   * ScanMapFUnction implements MapFunction to process input SamzaSqlRelMessages into output
+   * SamzaSqlRelMessage, performing the table scan
+   */
   private static class ScanMapFunction implements MapFunction<KV<Object, Object>, SamzaSqlRelMessage> {
     // All the user-supplied functions are expected to be serializable in order to enable full serialization of user
     // DAG. We do not want to serialize samzaMsgConverter as it can be fully constructed during stream operator
     // initialization.
     private transient SamzaRelConverter msgConverter;
+    private transient MetricsRegistry metricsRegistry;
+    private transient SamzaHistogram processingTime; // milli-seconds
+    private transient Counter inputEvents;
+    private transient Counter queryInputEvents;
+
     private final String streamName;
     private final int queryId;
+    private final String queryLogicalId;
+    private final String logicalOpId;
 
-    ScanMapFunction(String sourceStreamName, int queryId) {
+    ScanMapFunction(String sourceStreamName, int queryId, String queryLogicalId, String logicalOpId) {
       this.streamName = sourceStreamName;
       this.queryId = queryId;
+      this.queryLogicalId = queryLogicalId;
+      this.logicalOpId = logicalOpId;
     }
 
     @Override
     public void init(Context context) {
       TranslatorContext translatorContext =
           ((SamzaSqlApplicationContext) context.getApplicationTaskContext()).getTranslatorContexts().get(queryId);
       this.msgConverter = translatorContext.getMsgConverter(streamName);
+      ContainerContext containerContext = context.getContainerContext();
+      metricsRegistry = containerContext.getContainerMetricsRegistry();
+      processingTime = new SamzaHistogram(metricsRegistry, logicalOpId, TranslatorConstants.PROCESSING_TIME_NAME);
+      inputEvents = metricsRegistry.newCounter(logicalOpId, TranslatorConstants.INPUT_EVENTS_NAME);
```

Comment:
Same as the output, If the queryInput events and Scan level inputEvents are going to be the same. I think it may be good idea to just remove the scan level input Events.

Reply:
done